packages/cli/src/services/orchestration/main/MultiMainSetup.ee.ts

```diff
@@ -63,41 +63,43 @@ export class MultiMainSetup extends SingleMainSetup {
 	}
 
 	private async checkLeader() {
-		if (!this.redisPublisher.redisClient) return;
-
 		const leaderId = await this.redisPublisher.get(this.leaderKey);
 
-		if (!leaderId) {
-			this.logger.debug('Leadership vacant, attempting to become leader...');
-			await this.tryBecomeLeader();
+		if (leaderId === this.id) {
+			this.logger.debug(`[Instance ID ${this.id}] Leader is this instance`);
+
+			config.set('multiMainSetup.instanceType', 'leader');
+
+			await this.redisPublisher.setExpiration(this.leaderKey, this.leaderKeyTtl);
 
 			return;
 		}
 
-		if (this.isLeader) {
-			this.logger.debug(`Leader is this instance "${this.id}"`);
+		if (leaderId && leaderId !== this.id) {
```

Comment:
Even more importantly here, where the current instance realizes it is not the leader it's not firing `this.emit('leadershipChange', this.id);`

This means that even if it fails to maintain the lock and another leader takes over, the triggers here will never be stopped.

Reply:
I may be missing something - but from what I understand `this.emit('leadershipChange', this.id);` is already fired by the follower who gained leadership.

- `leaderId === this.id` → leader remained leader, so mark as leader (redundant sanity check)
- `leaderId && leaderId !== this.id` → follower remained follower, so mark as follower (redundant sanity check)
- `!leaderId` → no leader, so attempt to set key
	- if successfully set key, mark as leader and extend lease and emit leadership change event
	- else, mark as follower

Maybe faster to discuss?

Reply:
I think the path for becoming a leader is somewhat ok, the problem is when the instance loses leadership.

All we do is set `config.set('multiMainSetup.instanceType', 'follower');` but we're not disabling the triggers.

Makes sense?